src/IO/WriteBufferFromS3.cpp

```diff
@@ -53,6 +55,12 @@ WriteBufferFromS3::WriteBufferFromS3(
     , max_single_part_upload_size(max_single_part_upload_size_)
 {
     allocateBuffer();
+
+    LOG_TRACE(log, "thread_pool_size = {}", thread_pool_size);
+    if (thread_pool_size != 1)
+    {
+        writing_thread_pool = std::make_shared<ThreadPool>(thread_pool_size > 0 ? thread_pool_size : 1024);
```

Comment:
???

Reply:
> Performance comparison between old version and new. Let's try to upload test.hits to some s3 installation and check the query duration and speed. Also I am interested in `dstat` statistics.

I can't show you significant performance boost because it is about 2-3% and experiment would require a constant TCP retransmission rate in order to reliably prove it. I shall add retries to `WriteBufferFromS3` (which is a good thing anyway) and show it on simple HTTP connection drops.